plugins/transforms/concatfields/src/main/java/org/apache/hop/pipeline/transforms/concatfields/ConcatFields.java

```diff
@@ -166,13 +167,32 @@ private Object[] concatFields(Object[] inputRowData) throws HopException {
   }
 
   private void concatField(
-      StringBuilder targetField, IValueMeta valueMeta, Object valueData, String nullString)
+      StringBuilder targetField,
+      IValueMeta valueMeta,
+      Object valueData,
+      String nullString,
+      String trimType)
       throws HopValueException {
 
     if (valueMeta.isNull(valueData)) {
       targetField.append(nullString);
     } else {
-      targetField.append(valueMeta.getString(valueData));
+      if (trimType != null) {
+        switch (trimType) {
+          case ConcatFieldsMeta.TRIM_TYPE_LEFT:
+            targetField.append(valueMeta.getString(valueData).replaceAll("^\\s+", ""));
+            break;
+          case ConcatFieldsMeta.TRIM_TYPE_RIGHT:
+            targetField.append(valueMeta.getString(valueData).replaceAll("\\s+$", ""));
+            break;
+          case ConcatFieldsMeta.TRIM_TYPE_BOTH:
+            targetField.append(valueMeta.getString(valueData).trim());
+            break;
+          default:
+            targetField.append(valueMeta.getString(valueData));
+            break;
+        }
+      }
```

Comment:
We have a function in Const -> `org.apache.hop.core.Const#trimToType`

Reply:
Good to know let me use it and fix the code